src/NzbDrone.Core/Qualities/QualityModelComparer.cs

```diff
@@ -16,17 +17,45 @@ public QualityModelComparer(Profile profile)
             _profile = profile;
         }
 
+        public int Compare(int left, int right, bool respectGroupOrder = false)
+        {
+            var leftIndex = _profile.GetIndex(left);
+            var rightIndex = _profile.GetIndex(right);
+
+            if (respectGroupOrder)
+            {
+                return leftIndex.CompareTo(rightIndex);
+            }
+
+            return Math.Floor(leftIndex).CompareTo(Math.Floor(rightIndex));
+        }
+
         public int Compare(Quality left, Quality right)
         {
-            int leftIndex = _profile.Items.FindIndex(v => v.Quality == left);
-            int rightIndex = _profile.Items.FindIndex(v => v.Quality == right);
+            return Compare(left, right, false);
+        }
+
+        public int Compare(Quality left, Quality right, bool respectGroupOrder)
+        {
+            var leftIndex = _profile.GetIndex(left);
+            var rightIndex = _profile.GetIndex(right);
+
+            if (respectGroupOrder)
+            {
+                return leftIndex.CompareTo(rightIndex);
+            }
 
-            return leftIndex.CompareTo(rightIndex);
+            return Math.Floor(leftIndex).CompareTo(Math.Floor(rightIndex));
         }
 
         public int Compare(QualityModel left, QualityModel right)
         {
-            int result = Compare(left.Quality, right.Quality);
+            return Compare(left, right, false);
+        }
+
+        public int Compare(QualityModel left, QualityModel right, bool respectGroupOrder)
+        {
+            int result = Compare(left.Quality, right.Quality, respectGroupOrder);
```

Comment:
> Is it valid to compare revision if quality is not exactly the same?

No, it doesn't make sense.

When comparing for ordering we're explicitly getting the "index" which is the group index + quality in group index, so we're ordering consistently and only comparing the revision/real if the quality is an exact match.